tests/sentry/integrations/slack/test_utils.py

```diff
@@ -103,3 +103,193 @@ def test_rate_limiting(self):
+
+        self.integration = install_slack(self.event.project.organization)
+        self.add_list_response(
+            "conversations",
+            [
+                {"name": "my-channel", "id": "m-c"},
+                {"name": "other-chann", "id": "o-c"},
+                {"name": "my-private-channel", "id": "m-p-c", "is_private": True},
+            ],
+            result_name="channels",
+        )
+        self.add_list_response(
+            "users",
+            [
+                {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+                {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+                {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+            ],
+            result_name="members",
+        )
+
+    # self.add_msg_response("other-chann", "o-c")
+    # self.add_list_response(
+    #     "users",
+    #     [
+    #         {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+    #         {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+    #         {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+    #     ],
+    #     result_name="members",
+    # )
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def tearDown(self):
+        self.resp.__exit__(None, None, None)
+
+    def add_list_response(self, list_type, channels, result_name="channels"):
+        self.resp.add(
+            method=responses.GET,
+            url="https://slack.com/api/%s.list" % list_type,
+            status=200,
+            content_type="application/json",
+            body=json.dumps({"ok": True, result_name: channels}),
+        )
+
+    def add_msg_response(self, channelname, channelid, result_name="channel"):
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.scheduleMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps(
+                {"ok": True, result_name: channelid, "scheduled_message_id": "Q1298393284"}
+            ),
+        )
+
+    def add_bad_msg_response(self):
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.scheduleMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps({"ok": False, "error": "channel_not_found"}),
+        )
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def run_valid_test(self, channel, expected_prefix, expected_id, timed_out):
+        assert (expected_prefix, expected_id, timed_out) == get_channel_id(
+            self.organization, self.integration, channel
+        )
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def test_valid_channel_selected_new(self):
+        self.add_msg_response("my-channel", "m-c")
+        # ^^ this adds the responses this code path will hit
```

Comment:
lol you can delete my comments in these, they were just to help explain what I was doing.